tikv_util: Building on FreeBSD fails

## Bug Report


```
% cd components/tikv_util
% cargo build --keep-going -Z unstable-options
warning: profile package spec `tirocks-sys` in profile `dev` did not match any packages
   Compiling tikv_util v0.1.0 (/usr/home/dveeden/dev/tikv/tikv/components/tikv_util)
error[E0252]: the name `io` is defined multiple times
   --> components/tikv_util/src/sys/cpu_time.rs:205:15
    |
196 |     use std::io;
    |         ------- previous import of the module `io` here
...
205 |     use std::{io, mem, time::Duration};
    |               ^^--
    |               |
    |               `io` reimported here
    |               help: remove unnecessary import
    |
    = note: `io` must be defined only once in the type namespace of this module

error[E0433]: failed to resolve: use of undeclared crate or module `winapi`
   --> components/tikv_util/src/sys/cpu_time.rs:208:9
    |
208 |     use winapi::{
    |         ^^^^^^ use of undeclared crate or module `winapi`

error[E0432]: unresolved import `scopeguard`
   --> components/tikv_util/src/sys/cpu_time.rs:207:9
    |
207 |     use scopeguard::defer;
    |         ^^^^^^^^^^ use of undeclared crate or module `scopeguard`

error[E0659]: `defer` is ambiguous
   --> components/tikv_util/src/sys/cpu_time.rs:250:9
    |
250 |         defer! {{
    |         ^^^^^ ambiguous name
    |
    = note: ambiguous because of a conflict between a `macro_rules` name and a non-`macro_rules` name from another module
note: `defer` could refer to the macro defined here
   --> components/tikv_util/src/macros.rs:61:1
    |
61  | / macro_rules! defer {
62  | |     ($t:expr) => {
63  | |         let __ctx = $crate::DeferContext::new(|| $t);
64  | |     };
65  | | }
    | |_^
note: `defer` could also refer to the unresolved item imported here
   --> components/tikv_util/src/sys/cpu_time.rs:207:9
    |
207 |     use scopeguard::defer;
    |         ^^^^^^^^^^^^^^^^^
    = help: use `self::defer` to refer to this unresolved item unambiguously

warning: unused import: `io`
   --> components/tikv_util/src/sys/cpu_time.rs:205:15
    |
205 |     use std::{io, mem, time::Duration};
    |               ^^
    |
    = note: `#[warn(unused_imports)]` on by default

error[E0308]: mismatched types
   --> components/tikv_util/src/config.rs:658:33
    |
658 |         if fd_limit.rlim_cur >= expect {
    |                                 ^^^^^^ expected `i64`, found `u64`
    |
help: you can convert a `u64` to an `i64` and panic if the converted value doesn't fit
    |
658 |         if fd_limit.rlim_cur >= expect.try_into().unwrap() {
    |                                       ++++++++++++++++++++

error[E0308]: mismatched types
   --> components/tikv_util/src/config.rs:663:29
    |
663 |         fd_limit.rlim_cur = expect;
    |         -----------------   ^^^^^^ expected `i64`, found `u64`
    |         |
    |         expected due to the type of this binding

error[E0308]: mismatched types
   --> components/tikv_util/src/config.rs:664:32
    |
664 |         if fd_limit.rlim_max < expect {
    |                                ^^^^^^ expected `i64`, found `u64`
    |
help: you can convert a `u64` to an `i64` and panic if the converted value doesn't fit
    |
664 |         if fd_limit.rlim_max < expect.try_into().unwrap() {
    |                                      ++++++++++++++++++++

error[E0308]: mismatched types
   --> components/tikv_util/src/config.rs:666:33
    |
666 |             fd_limit.rlim_max = expect;
    |             -----------------   ^^^^^^ expected `i64`, found `u64`
    |             |
    |             expected due to the type of this binding

error[E0425]: cannot find function `processor_numbers` in this scope
   --> components/tikv_util/src/sys/cpu_time.rs:311:25
    |
311 |         let cpu = cpu * processor_numbers()? as u64;
    |                         ^^^^^^^^^^^^^^^^^ not found in this scope

Some errors have detailed explanations: E0252, E0308, E0425, E0432, E0433, E0659.
For more information about an error, try `rustc --explain E0252`.
warning: `tikv_util` (lib) generated 1 warning
error: could not compile `tikv_util` due to 9 previous errors; 1 warning emitted
```

--- a/components/tikv_util/src/config.rs
+++ b/components/tikv_util/src/config.rs
@@ -647,6 +647,9 @@ pub fn ensure_dir_exist(path: &str) -> Result<(), Box<dyn Error>> {
 
 #[cfg(unix)]
 pub fn check_max_open_fds(expect: u64) -> Result<(), ConfigError> {
+    #[cfg(target_os = "freebsd")]
+    let expect = expect as i64;
+
     use std::mem;
 
     unsafe {

--- a/components/tikv_util/src/sys/cpu_time.rs
+++ b/components/tikv_util/src/sys/cpu_time.rs
@@ -83,7 +83,7 @@ impl ProcessStat {
     }
 }
 
-#[cfg(target_os = "linux")]
+#[cfg(any(target_os = "linux", target_os = "freebsd"))]
 mod imp {
     use std::{fs::File, io, io::Read, time::Duration};
 
@@ -191,7 +191,7 @@ mod imp {
     }
 }
 
-#[cfg(not(any(target_os = "linux", target_os = "macos")))]
+#[cfg(target_os = "windows")]
 mod imp {
     use std::io;
 

